src/assets/scripts/aircraft/Waypoint.js

```diff
@@ -14,61 +63,167 @@ import _has from 'lodash/has';
      */
     parse(data, fms) {
+        this.extractFixRestrictions(data);
+
+        this.route = _get(data, 'route', this.route);
+
         // Populate Waypoint with data
         if (data.fix) {
-            this.navmode = 'fix';
+            this.navmode = WAYPOINT_NAV_MODE.FIX;
             this.fix = data.fix;
-            this.location = window.airportController.airport_get().getFixPosition(data.fix);
+            this.location = FixCollection.getFixPositionCoordinates(data.fix);
+        }
+
+        this.setInitialNavMode();
+    }
+
+    /**
+     * @for Waypoint
+     * @method extractFixRestrictions
+     * @param fixRestrictions {object}
+     */
+    extractFixRestrictions({ fixRestrictions }) {
+        if (_isNil(fixRestrictions)) {
+            return;
+        }
+
+        this.fixRestrictions = fixRestrictions;
+    }
+
+    /**
+     * If there isn't a navmode set, set one here
+     *
+     * For aircraft that don't yet have proper guidance (eg: SID/STAR, or departing aircraft)
+     *
+     * @for Waypoint
+     * @setInitialNavMode
+     */
+    setInitialNavMode() {
+        if (this.navmode) {
+            return;
+        }
+
+        this.navmode = WAYPOINT_NAV_MODE.HEADING;
+        const airport = window.airportController.airport_get();
+        const firstRouteSegment = _head(this.route.split('.'));
+
+        if (firstRouteSegment === airport.icao && this.heading === null) {
+            // aim departure along runway heading
+            const { angle } = airport.getRunway(airport.runway);
+
+            this.heading = angle;
+        } else if (firstRouteSegment === 'KDBG' && this.heading === null) {
+            // FIXME: radial is not defined or set anywhere in this class
+            // aim arrival @ middle of airspace
+            this.heading = this.radial + Math.PI;
+        }
+    }
+
+    // TODO: rename centerCeiling and make this method more flexible
+    /**
+     * @for Waypoint
+     * @method setAltitude
+     * @param centerCeiling {number}  ceiling of the airspace in feet
+     * @param cruiseAltitude {number} cruiseAltitude of the current aircraft
+     */
+    setAltitude(centerCeiling = null, cruiseAltitude) {
+        const { alt: altitudeRestriction } = this.fixRestrictions;
+
+        if (!altitudeRestriction) {
+            this.altitude = !_isNil(centerCeiling)
+                ? Math.min(centerCeiling, cruiseAltitude)
+                : cruiseAltitude;
+
+            return;
+        }
+
+        // TODO: there has to be an easier way to do this logic.
+        if (altitudeRestriction.indexOf(ABOVE_SYMBOL) !== INVALID_INDEX) {
+            // at-or-above altitudeRestriction restriction
+            const minAlt = parseInt(altitudeRestriction.replace(ABOVE_SYMBOL, ''), DECIMAL_RADIX);
+            const minimumAltitudeWithoutSymbol = minAlt * FL_TO_THOUSANDS_MULTIPLIER;
+
+            this.altitude = Math.min(minimumAltitudeWithoutSymbol, cruiseAltitude);
```

Comment:
@erikquinn is `Math.min` correct here?  I would thin it should be `cruiseAltitude` unless `minimumAltitudeWithoutSymbol` is higher.